src/metabase/mbql/schema.clj

```diff
@@ -273,7 +277,14 @@
    :else
    Field))
 
-(defclause ^{:requires-features #{:expressions}} +, x ExpressionArg, y ExpressionArg, more (rest ExpressionArg))
+(def ^:private ExpressionArgOrInterval
+  (s/if (partial is-clause? :interval)
+    interval
+    ExpressionArg))
+
+(defclause ^{:requires-features #{:expressions}} +
+  x ExpressionArg, y ExpressionArgOrInterval, more (rest ExpressionArg))
```

Comment:
Is there a reason to not allow intervals for the `more` args? Or to not allow it for `-` as well?

Reply:
I was a bit hesitant about multiple args because date arithemtics is a bit funky and say associativity property doesn't entirely hold. But that might be more a theoretical than practical concern. Happy to add if you reckon.

As for `-` I suppose it makes more sense to have it, than to strive for maximal parsimony